packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -1545,30 +1543,22 @@ export class BaseResolversVisitor<
       };
 
       if (this._federation.isResolveReferenceField(node)) {
-        if (this.config.generateInternalResolversIfNeeded.__resolveReference) {
-          const federationDetails = checkObjectTypeFederationDetails(
-            parentType.astNode as ObjectTypeDefinitionNode,
-            this._schema
-          );
-
-          if (!federationDetails || federationDetails.resolvableKeyDirectives.length === 0) {
-            return '';
-          }
-          signature.modifier = ''; // if a federation type has resolvable @key, then it should be required
+        if (!this._federation.getMeta()[parentType.name].hasResolveReference) {
+          return { value: '', meta };
         }
-
-        this._federation.setMeta(parentType.name, { hasResolveReference: true });
```

Comment:
We no longer check and collect meta as we hit federation fields any more because we are doing it before running visitor. [Relevant comment](https://github.com/dotansimha/graphql-code-generator/pull/10221/files?diff=split&w=1#r1912379037).
Therefore, we can check federation meta directly here